### PR TITLE
Support more types

### PR DESCRIPTION
**Pull request by farodin91**
Based on error_chain PR #19

### CODE REVIEW

**Comment by oli-obk** [comment]
Can you rebase this so all the noise from the error_chain PR is removed?

**Comment by farodin91** [comment]
Yes

File: src/lib.rs [hunk 1]
```diff
@@ -142,20 +142,25 @@ impl<'de, R: Read> Deserializer<R> {
...
         F: FnOnce(N) -> Result<V::Value>,
     {
+        if let XmlEvent::StartElement { .. } = *self.peek()? {
+            self.set_map_value()
+        }
         self.read_inner_value::<V, _>(|this| {
             if let XmlEvent::EndElement { .. } = *this.peek()? {
-                return Err(ErrorKind::Custom(format!("expected an integer")).into());
+                return Err(
+                    ErrorKind::Custom("Doesn't expected end element".to_string()).into(),
```
Comment (oli-obk) [inline src/lib.rs]: That's fairly obvious. Maybe state that a number was expected?

Comment (farodin91) [inline src/lib.rs]: It could be a number or boolean

Comment (oli-obk) [inline src/lib.rs]: Right. Just list both of these.

Comment (farodin91) [inline src/lib.rs]: I found a solution

Comment (farodin91) [inline src/lib.rs]: done

**Pull request merged**